### ISSUE TITLE
🐛 Bug Report:  provider is not configured to supported sign-in

### ISSUE THREAD

**Issue by AnishAnil** [backstage/backstage#25791]
### 📜 Description

Install the latest version of backstage
Follow instructions to configure the authentication for Azure ADFS --> https://backstage.io/docs/auth/microsoft/provider

Make changes to the front end and app-config.yaml

Add the values to in packages/backend/src/index.ts 

Add the changes to the front end --> packages/app/src/App.tsx

Restart backstage and it fails with 

![image](https://github.com/user-attachments/assets/121e6891-b523-41ac-ac14-f6318fe1da11)

### 👍 Expected behavior

The Authentication should go through without any issues

### 👎 Actual Behavior with Screenshots

![techshelf_error](https://github.com/user-attachments/assets/b65aac4c-d40e-4873-b91d-16b018ab3456)

### 👟 Reproduction steps

I'm following the documentation and the steps mentioned in the notes

### 📃 Provide the context for the Bug.

The older files structure where we had /packages/backend/src/plugibs/auth.tsx where we enable the authentication is no more available.  Without these files the authentication fails

### 🖥️ Your Environment

```
yarn run v1.22.22
$ /adfs_latest/backstage/node_modules/.bin/backstage-cli info
OS:   Linux 5.15.0-1044-aws - linux/x64
node: v20.13.1
yarn: 1.22.22
cli:  0.26.11 (installed)
backstage:  1.29.0

Dependencies:
  @backstage/app-defaults                                          1.5.9
  @backstage/backend-app-api                                       0.8.0
  @backstage/backend-common                                        0.23.3
  @backstage/backend-defaults                                      0.4.1
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.15
  @backstage/backend-plugin-api                                    0.7.0
  @backstage/backend-tasks                                         0.5.27
  @backstage/catalog-client                                        1.6.5
  @backstage/catalog-model                                         1.5.0
  @backstage/cli-common                                            0.1.14
  @backstage/cli-node                                              0.2.7
  @backstage/cli                                                   0.26.11
  @backstage/config-loader                                         1.8.1
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.14.1
  @backstage/core-compat-api                                       0.2.7
  @backstage/core-components                                       0.14.9
  @backstage/core-plugin-api                                       1.9.3
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.8
  @backstage/frontend-plugin-api                                   0.6.7
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.29
  @backstage/integration                                           1.13.0
  @backstage/plugin-api-docs                                       0.11.7
  @backstage/plugin-app-backend                                    0.3.71
  @backstage/plugin-app-node                                       0.1.22
  @backstage/plugin-auth-backend-module-atlassian-provider         0.2.3
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.14
  @backstage/plugin-auth-backend-module-azure-easyauth-provider    0.1.5
  @backstage/plugin-auth-backend-module-bitbucket-provider         0.1.5
  @backstage/plugin-auth-backend-module-cloudflare-access-provider 0.1.5
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.17
  @backstage/plugin-auth-backend-module-github-provider            0.1.19
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.19
  @backstage/plugin-auth-backend-module-google-provider            0.1.19
  @backstage/plugin-auth-backend-module-guest-provider             0.1.8
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.17
  @backstage/plugin-auth-backend-module-oauth2-provider            0.2.3
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.15
  @backstage/plugin-auth-backend-module-oidc-provider              0.2.3
  @backstage/plugin-auth-backend-module-okta-provider              0.0.15
  @backstage/plugin-auth-backend-module-onelogin-provider          0.1.3
  @backstage/plugin-auth-backend                                   0.22.9
  @backstage/plugin-auth-node                                      0.4.17
  @backstage/plugin-auth-react                                     0.1.4
  @backstage/plugin-bitbucket-cloud-common                         0.2.21
  @backstage/plugin-catalog-backend-module-logs                    0.0.1
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.20
  @backstage/plugin-catalog-backend                                1.24.0
  @backstage/plugin-catalog-common                                 1.0.25
  @backstage/plugin-catalog-graph                                  0.4.7
  @backstage/plugin-catalog-import                                 0.12.1
  @backstage/plugin-catalog-node                                   1.12.4
  @backstage/plugin-catalog-react                                  1.12.2
  @backstage/plugin-catalog                                        1.21.1
  @backstage/plugin-events-node                                    0.3.8
  @backstage/plugin-org                                            0.6.27
  @backstage/plugin-permission-backend-module-allow-all-policy     0.1.19
  @backstage/plugin-permission-backend                             0.5.46
  @backstage/plugin-permission-common                              0.8.0
  @backstage/plugin-permission-node                                0.8.0
  @backstage/plugin-permission-react                               0.4.24
  @backstage/plugin-proxy-backend                                  0.5.3
  @backstage/plugin-scaffolder-backend-module-azure                0.1.14
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.12
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.12
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.12
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.14
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.12
  @backstage/plugin-scaffolder-backend-module-github               0.4.0
  @backstage/plugin-scaffolder-backend-module-gitlab               0.4.4
  @backstage/plugin-scaffolder-backend                             1.23.0
  @backstage/plugin-scaffolder-common                              1.5.4
  @backstage/plugin-scaffolder-node                                0.4.8
  @backstage/plugin-scaffolder-react                               1.10.0
  @backstage/plugin-scaffolder                                     1.23.0
  @backstage/plugin-search-backend-module-catalog                  0.1.28
  @backstage/plugin-search-backend-module-pg                       0.5.32
  @backstage/plugin-search-backend-module-techdocs                 0.1.27
  @backstage/plugin-search-backend-node                            1.2.27
  @backstage/plugin-search-backend                                 1.5.14
  @backstage/plugin-search-common                                  1.2.13
  @backstage/plugin-search-react                                   1.7.13
  @backstage/plugin-search                                         1.4.14
  @backstage/plugin-signals-react                                  0.0.4
  @backstage/plugin-techdocs-backend                               1.10.9
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.12
  @backstage/plugin-techdocs-node                                  1.12.8
  @backstage/plugin-techdocs-react                                 1.2.6
  @backstage/plugin-techdocs                                       1.10.7
  @backstage/plugin-user-settings-common                           0.0.1
  @backstage/plugin-user-settings                                  0.8.10
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.5.9
  @backstage/theme                                                 0.5.6
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.8
Done in 0.63s.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by awanlin** [comment]
Hi @AnishAnil, can you share the changes you made? There could be a small typo or formatting that could be causing this so seeing this would help identify it.

**Comment by AnishAnil** [comment]
@awanlin , @What has changed in the new version of techshelf. I'm unable to find the plugins directory where we specify the auth.ts file. Even after installing multiple plugins for backend..........the plugin directory is not available for making any modification.

Files that I modified

![image](https://github.com/user-attachments/assets/e7deefd4-eec6-4fe8-aeb6-95822119673e)


![image](https://github.com/user-attachments/assets/0c8707ba-cbbc-4c6b-92c1-1854da2ee9ca)

![image](https://github.com/user-attachments/assets/35ac2910-373e-4826-8e5c-340ef9d2e914)
![image](https://github.com/user-attachments/assets/5f70b1bc-d5c8-4d66-be0e-ade0c9ffdb16)
![image](https://github.com/user-attachments/assets/ab999faf-9da4-4c74-b6cf-600bfe87e875)


This is a simple issue to replicate. Just install a Vanila techshelf and any plugin like bitbucket which should automatically generate a plugin directory and it will simply not populate.

Thank you,
Anish

**Comment by anishaanil1** [comment]
Hello @awanlin, Do you have some direction to resolve this issue. Today we had a call with the Azure ADFS team and it looks like the application needs more permissions than what has been requested in your document.

![image](https://github.com/user-attachments/assets/ea306523-0a0b-49f5-8990-2083612812d8)


Why is the application requesting for --> Send E-Mail as you permission which is not part of the documentation.

We are currently stuck and unable to proceed. Any help is greatly appreciated.

Thank you

**Comment by Rugvip** [comment]
This might be fixed by #25728, or at least provide some more context for the issue

**Comment by jpitapeva** [comment]
Hi, i did the install today for 1.30 version by (npx @backstage/create-app@latest) and show the same error, can you help me?

Error
`Login failed; caused by Error: The microsoft provider is not configured to support sign-in`

---

file: packages/app/src/App.tsx

import { microsoftAuthApiRef } from '@backstage/core-plugin-api';
```
components: {
    SignInPage: props => <SignInPage {...props} 
    auto providers={[
      {
        id: 'microsoft-auth-provider',
        title: 'microsoft',
        message: 'Sign in using microsoft',
        apiRef: microsoftAuthApiRef,
      },
    
    ]}     
```

---

file: packages/backend/src/index.ts

```backend.add(import('@backstage/plugin-auth-backend-module-microsoft-provider'));```


---

file: app-config.yaml

```
auth:
  environment: development
  providers:
    microsoft:
      development:
        clientId: "REMOVED"
        clientSecret: "REMOVED"
        tenantId: "REMOVED"
        domainHint: "REMOVED"
        signIn:
           resolvers:
             # typically you would pick one of these
             - resolver: emailMatchingUserEntityProfileEmail
             - resolver: emailLocalPartMatchingUserEntityName
             - resolver: emailMatchingUserEntityAnnotation
```

---

Tutorial:
https://backstage.io/docs/auth/microsoft/provider

**Comment by awanlin** [comment]
Hi @jpitapeva, I can't test this end to end as I don't have an Azure AD account but the one step that is missing here is adding the package. Did you run: `yarn --cwd packages/backend add @backstage/plugin-auth-backend-module-microsoft-provider` as per the Azure docs here: https://backstage.io/docs/auth/microsoft/provider#backend-installation

**Comment by jpitapeva** [comment]
Hi @awanlin , I instaled `yarn --cwd packages/backend add @backstage/plugin-auth-backend-module-microsoft-provider` but the problem persist

**Comment by jpitapeva** [comment]
Hi @AnishAnil, can you retest with the last backstage version?

**Comment by jpitapeva** [comment]
Hi @awanlin, i did an test using my ad microsft and backstage 1.16 this version work

Code
https://github.com/foster-devops/backstage

**Comment by AnishAnil** [comment]
@jpitapeva , The Azure ADFS and Microsoft authentication works differently, So not sure if we are hitting the same issue. 
It was working perfectly until we upgraded to the latest version. The new version asks for additional permission --> "“Send mail as you” and again this is a violation and most companies will not allow it. So we need it working as it was in older versions

**Comment by jpitapeva** [comment]
@AnishAnil , in backstage 1.6 version for example it's working

**Comment by AnishAnil** [comment]
It fails with the same error on the latest version aswell
{
  "version": "1.30.0"
}

**Comment by awanlin** [comment]
@AnishAnil and @jpitapeva, can you run this - `yarn backstage-cli config:check --strict` - and share the results please? This error is most common when there is an issue with the config and the results of this command might help us.

**Comment by jpitapeva** [comment]
Hi @awanlin, executed the command `yarn backstage-cli config:check --strict`

```
$ yarn backstage-cli config:check --strict
yarn run v1.22.19
$ D:\Source\backstagetrinta\node_modules\.bin\backstage-cli config:check --strict
Loaded config from app-config.yaml
Done in 3.02s.
```

**Comment by AnishAnil** [comment]
> @AnishAnil and @jpitapeva, can you run this - `yarn backstage-cli config:check --strict` - and share the results please? This error is most common when there is an issue with the config and the results of this command might help us.

![image](https://github.com/user-attachments/assets/87683873-5a4c-4678-9b76-55abaf2c55b4)

**Comment by AnishAnil** [comment]
I have made a clear document so that can be reviewed. The changes from the older version to the lastest version.
[AzureADFS.txt](https://github.com/user-attachments/files/16705878/AzureADFS.txt)


Kindly refer to the document.
Also, an OLD Article pointing to how it was working on earlier version.

https://github.com/backstage/backstage/issues/11368

Can someone please test and provide a fix. Our entire stack is blocked with this upgrade

**Comment by freben** [comment]
Can you remove the entire `signIn` block from your config please? I see you are supplying a custom one in code, and perhaps unintuitively, the config based resolvers take precedence since https://backstage.io/docs/releases/v1.30.0#breaking-auth-sign-in-resolver-priority

**Comment by AnishAnil** [comment]
> `signIn`

But from which file. The text file i added is for older versions that i configured and the latest version there is none of these changes. Infact we tried it with bitbucket Authentication and that also fails with the similar errors

**Comment by jpitapeva** [comment]
Hi @freben, the backstage 1.6 version it's working for me

source code
https://github.com/foster-devops/backstage

**Comment by AnishAnil** [comment]
@freben , Is there a direction for us ...............We are currently blocked n there are multiple users facing the same issue.

**Comment by freben** [comment]
OK let me just start by saying that in the new backend system, there's not meant to be any `plugins` folder. You could make one for your own needs or so, but most backends will start out with only an `index.ts` and that's intentional. Almost all plugins and modules are a single `backend.add` and then possibly some config. So the initial questions about no plugins folder being made - that can be put aside. Either way, that was only a convention, not a framework thing or rule; the structure of the folders doesn't really matter.

Regarding the error. I just made a fresh backstage with the module added, and on the sign in page, clicking the "Sign in" button opens the popup as expected at least. But I also do not have any Azure account to lean against. Did you get the error immediately when clicking Sign in, or only after completing the sign in flow?

**Comment by jpitapeva** [comment]
Hello @freben, attached shows the error, this error happens after I enter my credentials
![NO_RESOLVERS](https://github.com/user-attachments/assets/52d04d4e-bac3-4679-a202-16e14dcbcd4f)


example without signing and without plugin custom

app-config.yaml
auth:
  providers:
    microsoft:
      development:
        clientId: "REMOVED"
        clientSecret: "qcC8QREMOVED"
        tenantId: "REMOVED"
        domainHint: "REMOVED"

**Comment by benjdlambert** [comment]
OK, so the last error message is progress here. Its now saying that the user couldn't be found in the Catalog. Are you ingesting users from Entra ID or something into the catalog?

The idea here is that you should pick one of the resolvers, and that then picks how to resolve the user from the catalog.
https://backstage.io/docs/auth/microsoft/provider#resolvers

You can also follow the above instructions to implement your own resolver if you don't have users in the catalog and don't plan to.

**Comment by AnishAnil** [comment]
@benjdlambert , I have personally followed every line in the document and checked and it fails, We additionally got the Azure team involved to check on this issue, and as per them there is a variation in permissions between the documentation provided by backstage and the requested permissions.

Permissions requested by backstage as per the documentation:

![image](https://github.com/user-attachments/assets/442e7dc7-78e6-4f6c-bdf8-bd99d51029ef)


However, when we checked in Azure ADFS logs it ask's for addional permissions which is  Send mail as you. 
As per any security rules of the Organization this will not be allowed.

![355810574-e4f065b2-8e61-44b4-8ace-312b1f708356](https://github.com/user-attachments/assets/57e42ef5-71a2-4799-bdf5-c3579bcd2b62)

Kindly check  as this is a breaking the setup.

**Comment by benjdlambert** [comment]
@AnishAnil, sorry, let me clarify a little bit. This is nothing to do with permissions. When you added the `signInResolvers` block, this is basically saying that the user that you're trying to sign in as does not exist in the catalog, and the resolvers that you have listed all look inside the Software Catalog for a user.

So do you have any users already ingested into your catalog? If not then you will have to create your own resolver which can issue a token.

**Comment by AnishAnil** [comment]
Without the signInResolvers block it does not even give you the Azure option.

 It simply allows to to sign in as Guest which is not an expected behavior.

**Comment by benjdlambert** [comment]
@AnishAnil you're still going to keep the microsoft part, you're just going to implement your own resolver, although you still haven't answered if you have any users in the Backstage Software Catalog.

Please read though https://backstage.io/docs/auth/identity-resolver/#building-custom-resolvers especially https://backstage.io/docs/auth/identity-resolver/#sign-in-without-users-in-the-catalog if you don't have any users in the catalog ingested from Entra ID.

**Comment by benjdlambert** [comment]
> We are not adding any users to our catalog.

So if you're not adding any users to your catalog then you will have to follow the second link to create a custom `signInResolver` that will get the response from Azure sign-in and then you need to return a token for the user.

The traditional way is to import the users into the Catalog, and then pick a `signInResolver` which can find the user entity in the Catalog for you rather than writing your own `signInResolver`, but if you're not planning on importing the users then you'll need to write your own.

You should be able to adapt the code from https://backstage.io/docs/auth/identity-resolver/#building-custom-resolvers to the microsoft resolver instead, and then the resolver implementation might look something like https://backstage.io/docs/auth/identity-resolver/#sign-in-without-users-in-the-catalog

**Comment by AnishAnil** [comment]
We tried everything and it simply fails........................with the same error.

**Comment by jpitapeva** [comment]
Hi @benjdlambert , the backstage 1.6 version it's working using the same microsoft EntraId

source code
https://github.com/foster-devops/backstage

**Comment by benjdlambert** [comment]
@AnishAnil can I see your custom resolver? And the config that you're using? And that you're adding it to `packages/backend`?

**Comment by jpitapeva** [comment]
@benjdlambert , Hi

code
https://github.com/jpitapeva/backstage-auth-test/blob/master/app-config.yaml

**Comment by AnishAnil** [comment]
@jpitapeva , @benjdlambert , I have the same lines but just the configuration is different. Only two files are edited in the new setup:

1> app-config.yaml
2> homedir/package/src/App.tsx --> Just add the microsoft block else we will just get the Guest User block.

Honestly, I can share a screen and take you through it than juggling around. :-)

Rest all the files are just defaults.

**Comment by benjdlambert** [comment]
There's no custom resolver in any of those examples which means it's going to try and use the catalog to get the user details, but neither of you have any users in the Catalog. You need to create your own custom resolver instead of using the ones defined in config. Please read through the documentation paying close attention to the part where you have to write your own resolver. https://backstage.io/docs/auth/identity-resolver/#sign-in-without-users-in-the-catalog

**Comment by AnishAnil** [comment]
IMO, and on a very polite n humble thought, 

we had a working solution, which is now broken. The same integration works with GitHub Authentication. Why no custom resolver needs to be created????? To be honest, There is a gap that doesn't look like a clear understanding of the problem by the backstage team.

Just to keep you informed. We have had a very old version of techshelf test instance which we upgraded to the latest version and it maintained the directory structure with the plugin directory and it is still working. So Clearly, the team is broken something in the latest version

Can the backstage team actually test and come up with a working solution or simply tell Azure Authentication will not work. 
Or create a clear documentation on how to get your docuemnation with Azure ADFS to work?????

**Comment by benjdlambert** [comment]
I'm sorry you're having problems, and I can understand your frustration of things breaking between the monthly releases of Backstage. I believe that you're being hit by some of the breaking changes that we made in order to enable security by default, so that now all requests that go through backstage have to be authorized in some way, and that means that you have to pick a way to 'Sign-In' to Backstage.

We've tried to call out some of the changes in the release notes in [previous](https://backstage.io/docs/releases/v1.24.0) [versions](https://backstage.io/docs/releases/v1.26.0), as well as [updating the docs](https://backstage.io/docs/auth/identity-resolver#backstage-user-identity) where necessary, but maybe we could have done more to make these changes more visible.

I know that there are adopters today using Azure Sign-In successfully after the upgrade, so I'm not sure that we've broken things, it's just that there's other things that now need to be done in order to keep Backstage safe and secure.

**Comment by AnishAnil** [comment]
@benjdlambert , I disagree that I'm the only customer facing this issue, As you can see there is already another person @jpitapeva ,  in the same ticket having the problem. I also remember another ticket being raised for an authentication issue with Azure with similar errors............So there are definitely more than 2 people seeing similar behavior Unless the team really wants to disagree. :-). If needed, I can offer you a webex session to run through our setup or collect any relevant data to fix this issue.

**Comment by benjdlambert** [comment]
> I disagree that I'm the only customer facing this issue

I'm not sure that's what I said - I said that there are adopters today that have gotten this setup and working on the latest versions with either users in the catalog or custom resolvers.

@jpitapeva after writing a custom resolver or importing users into the Software Catalog, does sign in work for you?

>  I can offer you a webex session to run through our setup or collect any relevant data to fix this issue.

I'm not sure that a webex session is necessary in this case, this seems like expected and documented behaviour, and there's documentation to follow in order to create custom sign-in resolvers or populate the catalog with users after the auth changes. And again like I said before, if we can make this clearer that this is the required path forward, then I'm happy to revisit some of the documentation to help others that might be facing similar issues.

**Comment by jpitapeva** [comment]
@benjdlambert I don't create custom resolver, in backstage 1.16 version i never used before custom

**Comment by benjdlambert** [comment]
@jpitapeva  Yes, If a new user enters your team, then they will need to be imported into the Catalog in order to allow them to Sign-In. We recommend setting up some ingestion into the Catalog that can pull these users into the Catalog from an authoritative source,  Entra ID / LDAP / Active Directory or what not in order to get the users into the catalog that are allowed to Sign-In to Backstage.

**Comment by Rugvip** [comment]
Sorry but the discussion in this issue has diverged far from the original report, which has been resolved. I'm going to close this one, but please open new issues for any outstanding problems so that we can discuss them in isolation.

**Comment by AnishAnil** [comment]
@Rugvip, Why would you close this ticket.................. What makes you feel that it is heading in the wrong direction?

Clearly, we have a problem statement that multiple users are unable to login to Backstage when configured with Azure ADFS. This is working on an older version and not on the newer version. 

This is a clear defect and I would request you not close this issue
